### ISSUE TITLE
Video File Viewer does not work with Amazon S3 signed urls

### ISSUE THREAD

**Issue by kunningham** [Telligent/Amazon-S3-CFS-Provider#1]
IFileViewer plugin (VideoFileViewer) does not play video from S3 when url is a signed url.

The root cause for strange/inconsistent behavior of video files from S3 to be the following: Telligent uses html5 video tag to stream videos. Video source is specified using <source> tag which contain src and type. Type means MIME type and it should be "video/mp4" for mp4 videos. But when videos are uploaded to S3, type is set as "application/octet-stream". Internet Explorer is able to resolve this MIME type and play the video whereas Chrome/ Firefox is not able process this MIME type.

Analyzing further, Telligent uses an instance of IFileViewer plugin (VideoFileViewer) to render videos. This derives MIME type from file extension. Following the normal convention for files, they pick everything after the last dot (.) as file extension. As S3 url is signed url, it will have access key id, expiry and signature after extension. The Telligent plugin does not take this in to account, extension is not derived properly and that result in wrong MIME type.

**Comment by afscrome** [comment]
This sounds like a problem I faced with the Azure Blob Storage provider.

The mime type of "application/octet-stream" is probably just a default in the basense of explicitly setting a mime type when the file is originally uploaded.  

https://github.com/afscrome/TelligentAzure/blob/master/Filestorage/AzureBlobFilestorageProvider.cs#L117 is the fix for the azure provider.

**Comment by afscrome** [comment]
Looks like we are already doing this https://github.com/Telligent/Amazon-S3-CFS-Provider/blob/8a22d88c6802595c2e844be116d974890dec5fe4/src/AmazonS3FileStorageProvider.cs#L203 .

**Comment by bentiedt** [comment]
Yes. That should prevent the issue for new files. I don't think that was always present, though, so its possible there are older files that have incorrect MIME types within S3 (or perhaps files were uploaded before the appropriate MIME types were registered in the platform). In any case, the resolution may be to enable a global HEAD review of all files via a background job.
